### PR TITLE
The DigitalIdentity interface should be renamed DigitalCredential

### PR DESCRIPTION
**Pull request by marcoscaceres**
#### 86a7cc9223d3f1634215b1e48dac48b07407d79f
<pre>
The DigitalIdentity interface should be renamed DigitalCredential
<a href="https://bugs.webkit.org/show_bug.cgi?id=270197">https://bugs.webkit.org/show_bug.cgi?id=270197</a>
<a href="https://rdar.apple.com/123747961">rdar://123747961</a>

Reviewed by Anne van Kesteren.

Renamed interface to reflect spec change:
<a href="https://github.com/WICG/digital-identities/pull/79/">https://github.com/WICG/digital-identities/pull/79/</a>

* LayoutTests/http/wpt/credential-management/setDigitalIdentityEnable.https.html:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/credentialmanagement/BasicCredential.cpp:
(WebCore::BasicCredential::type const):
* Source/WebCore/Modules/credentialmanagement/BasicCredential.h:
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.cpp:
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.h:
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.cpp: Renamed from Source/WebCore/Modules/credentialmanagement/DigitalIdentity.cpp.
(WebCore::DigitalCredential::create):
(WebCore::DigitalCredential::DigitalCredential):
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.h: Renamed from Source/WebCore/Modules/credentialmanagement/DigitalIdentity.h.
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.idl: Renamed from Source/WebCore/Modules/credentialmanagement/DigitalIdentity.idl.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/JSBasicCredentialCustom.cpp:
(WebCore::toJSNewlyCreated):
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:

Canonical link: <a href="https://commits.webkit.org/275442@main">https://commits.webkit.org/275442@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/0f4b484a34580f2ed688e0c539d0403889f6103c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41840 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20854 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44222 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44422 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37935 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/44147 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/24009 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18185 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34577 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42414 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17783 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36029 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15248 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15471 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/37053 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45818 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/38026 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37374 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/41130 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16655 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13674 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39578 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18274 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/36330 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9377 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18333 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17918 "Built successfully") | | |

### CODE REVIEW

**Comment by annevk** [comment]
Presumably you also want to change the interface name from `DigitalIdentity`? Also seems like you forgot to update all the files that reference these files.

File: Source/WebCore/WebCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -38809,7 +38809,7 @@
 				938430B42475DE40003F9617 /* DictationContext.h in Headers */,
 				937FF3D51A1012D6008EBA31 /* DictionaryLookup.h in Headers */,
 				2D5646B01B8F8493003C4994 /* DictionaryPopupInfo.h in Headers */,
-				0D431A7D2B5A397C00864655 /* DigitalIdentity.h in Headers */,
+				0D431A7D2B5A397C00864655 /* DigitalCredential.h in Headers */,
```
Comment (annevk) [inline Source/WebCore/WebCore.xcodeproj/project.pbxproj:38812]: This should be sorted.

**Comment by marcoscaceres** [comment]
> Presumably you also want to change the interface name from DigitalIdentity? Also seems like you forgot to update all the files that reference these files.

Yeah, I didn't `git add` on main before doing `git webkit pr`, so it only brought in the renames :/ .... should be all good now.

File: Source/WebCore/DerivedSources-input.xcfilelist [hunk 2]
```diff
@@ -437,7 +437,7 @@ $(PROJECT_DIR)/Modules/credentialmanagement/CredentialMediationRequirement.idl
 $(PROJECT_DIR)/Modules/credentialmanagement/CredentialRequestOptions.idl
 $(PROJECT_DIR)/Modules/credentialmanagement/CredentialsContainer.idl
 $(PROJECT_DIR)/Modules/credentialmanagement/DigitalCredentialRequestOptions.idl
-$(PROJECT_DIR)/Modules/credentialmanagement/DigitalIdentity.idl
+$(PROJECT_DIR)/Modules/credentialmanagement/DigitalCredential.idl
```
Comment (annevk) [inline Source/WebCore/DerivedSources-input.xcfilelist:439]: I think this needs to be one higher now.

File: Source/WebCore/DerivedSources-output.xcfilelist [hunk 3]
```diff
@@ -843,8 +843,8 @@ $(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/JSDeviceOrientationOrMotionPermissi
 $(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/JSDeviceOrientationOrMotionPermissionState.h
 $(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/JSDigitalCredentialRequestOptions.cpp
 $(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/JSDigitalCredentialRequestOptions.h
-$(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/JSDigitalIdentity.cpp
-$(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/JSDigitalIdentity.h
+$(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/JSDigitalCredential.cpp
+$(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/JSDigitalCredential.h
```
Comment (annevk) [inline Source/WebCore/DerivedSources-output.xcfilelist:845]: These should also be slightly higher now I think.

**Review by annevk — APPROVED** [review]
Looks good modulo sorting.

**Pull request merged**